### PR TITLE
Puts outfits into category, uses TGUI list input, hotkey mode to TGUI list input

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
* Puts outfits into category
* uses TGUI list input
* hotkey mode to TGUI list input

It's annoying that outfits are not well sorted, and it gets more annoying when you have to find a specific one in the dump.
Now it uses TGUI that you can search, but also most outfits are categorised.

also, TGUI list input accepts hotkey mode value.

## Why It's Good For The Game
better debugging & management

## Testing Photographs and Procedure

image

image

hotkey press works well.

image

image

image

category works well too

## Changelog
:cl:
add: Equip outfit now uses TGUI.
code: Outfits are categorised.
code: TGUI list input now accepts hotkey mode value to open it at hotkey mode at default
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/ListInputModal.tsx [hunk 1]
```diff
@@ -12,13 +12,18 @@ type ListInputData = {
...
+  hotkey_mode: boolean;
 };
 
 export const ListInputModal = (_, context) => {
   const { act, data } = useBackend<ListInputData>(context);
-  const { items = [], message = '', init_value, large_buttons, timeout, title } = data;
+  const { items = [], message = '', init_value, large_buttons, timeout, title, hotkey_mode } = data;
   const [selected, setSelected] = useLocalState<number>(context, 'selected', items.indexOf(init_value));
-  const [searchBarVisible, setSearchBarVisible] = useLocalState<boolean>(context, 'searchBarVisible', items.length > 9);
+  const [searchBarVisible, setSearchBarVisible] = useLocalState<boolean>(
```
Comment (itsmeow) [inline tgui/packages/tgui/interfaces/ListInputModal.tsx:22]: uh is this not actually used anywhere?

**Review by itsmeow — COMMENTED** [review]
Why hotkeys though. It's worse than search imo

File: tgui/packages/tgui/interfaces/ListInputModal.tsx (hunk 1 above)
Comment (EvilDragonfiend) [reply tgui/packages/tgui/interfaces/ListInputModal.tsx:22]: It is the actual toggle for hotkey and serach modes

**Comment by EvilDragonfiend** [comment]
> Why hotkeys though. It's worse than search imo

Because you would generally want to get debug outfit by 2 key presses: d, enter. 
Search bar isn't good with it.

File: tgui/packages/tgui/interfaces/ListInputModal.tsx (hunk 1 above)
Comment (itsmeow) [reply tgui/packages/tgui/interfaces/ListInputModal.tsx:22]: Oh I'm stupid. I didn't see the line removal so I thought you were just adding it. Don't code review on no caffeine.

**Review by itsmeow — APPROVED** [review]
Good improvement of the menu. I would like a dedicated outfit browser/creator TGUI but for now this is quite nice and improves on the original while keeping the same format.

**Comment by EvilDragonfiend** [comment]
> Good improvement of the menu. I would like a dedicated outfit browser/creator TGUI but for now this is quite nice and improves on the original while keeping the same format.

image

eh, I noticed they had this... maybe porting would be better.

**Comment by EvilDragonfiend** [comment]
I got an idea to merge selective things in a management window. I will reopen this PR along with another PR to merge.

**Pull request closed without merging**